packages/components/src/navigation/stories/index.js

```diff
@@ -1,88 +1,131 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Navigation from '../';
+import NavigationBackButton from '../back-button';
+import NavigationPane from '../pane';
+import NavigationMenu from '../menu';
+import NavigationTitle from '../title';
 
 export default {
 	title: 'Components/Navigation',
 	component: Navigation,
 };
 
 const data = [
-	{ title: 'WooCommerce', slug: 'root', back: 'Dashboard' },
-	{ title: 'Home', slug: 'home', parent: 'root', menu: 'primary' },
+	{ title: 'Home', id: 'home' },
 	{
 		title: 'Analytics',
-		slug: 'analytics',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'analytics',
 	},
 	{
 		title: 'Orders',
-		slug: 'orders',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'orders',
 	},
 	{
 		title: 'Overview',
-		slug: 'overview',
+		id: 'overview',
 		parent: 'analytics',
 	},
 	{
 		title: 'Products report',
-		slug: 'products',
+		id: 'products',
 		parent: 'analytics',
 	},
 	{
 		title: 'All orders',
-		slug: 'all_orders',
+		id: 'all_orders',
 		parent: 'orders',
 	},
 	{
 		title: 'Payouts',
-		slug: 'payouts',
+		id: 'payouts',
 		parent: 'orders',
 	},
 	{
 		title: 'Settings',
-		slug: 'settings',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'settings',
 		menu: 'secondary',
 	},
 	{
 		title: 'Extensions',
-		slug: 'extensions',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'extensions',
 		menu: 'secondary',
 	},
 	{
 		title: 'General',
-		slug: 'general',
+		id: 'general',
 		parent: 'settings',
 	},
 	{
 		title: 'Tax',
-		slug: 'tax',
+		id: 'tax',
 		parent: 'settings',
 	},
 	{
 		title: 'My extensions',
-		slug: 'my_extensions',
+		id: 'my_extensions',
 		parent: 'extensions',
 	},
 	{
 		title: 'Marketplace',
-		slug: 'marketplace',
+		id: 'marketplace',
 		parent: 'extensions',
 	},
 ];
 
 function Example() {
-	return <Navigation data={ data } initial="home" />;
+	const [ active, setActive ] = useState( 'home' );
+
+	return (
+		<Navigation
+			activeId={ active }
+			items={ data }
+			rootTitle="WooCommerce Home"
+		>
+			{ ( { currentItems, currentLevel, parentItems, parentLevel } ) => {
+				return (
+					<NavigationPane>
+						<NavigationBackButton
+							onClick={ () =>
+								parentLevel
+									? setActive( parentItems[ 0 ].id )
+									: ( window.location =
+											'https://wordpress.com' )
+							}
```

Comment:
I think we can hide much of this logic. I don't think the consumer should worry about this much logic here and other places. I would prefer they just supply a data tree and call backs:

```js
<NavigationBackButton
	parent={ parent }
	exitText="Dashboard"
	exitUrl="http://wordpress.com"
/>
```
But then maybe we should just incorporate this as part of the main parent component.

Reply:
For animation purposes, I don't think we can have this as part of the main parent component.  I think it would be possible to include it in the `NavigationPane` component, but I'm also concerned that our use case in WooCommerce is pretty specific and will not match the needs of consumers.

Do we think that most consumers will have exit buttons?  That most will activate the parent's first child on click?  If yes to both of those, then maybe we can include it, but this does feel very specific to our purposes.

Reply:
>For animation purposes, I don't think we can have this as part of the main parent component.

In the interest of this PR, maybe best to remove then. There are several ways to animate this and I think it'll best be evaluated in a separate PR.

>Do we think that most consumers will have exit buttons?

Why don't we supply an `onBack` prop that takes a function with an id passed as the argument? I see what you're saying about it being a Woo specific use case at the top level. 

>That most will activate the parent's first child on click?

Yes, most definitely.

Reply:
>Why don't we supply an onBack prop that takes a function with an id passed as the argument? 

The problem with this is that we're assuming the callback only needs an ID, which works well for our example that only works on IDs, but it may also need access to the `onClick` or `url` params of the first item.

Reply:
@psealock One compromise here may be to pass the parent's first child by default as `backItem` if we're sure this is the item most consumers will want to navigate to.  This way it can still pass all the details and we can treat those like props in this component so it can define `url` or `onClick` behavior.

I think there may also be some other ways to simplify this by using context and passing information by default to child components, but prefer to handle in a follow-up.

What do you think?